### ISSUE TITLE
[EPIC][AGENTIC RSI] Scored subagent self-improvement with Agent Lightning + SkillOpt, local and remote

### ISSUE THREAD

**Issue by lost-rob0t** [lost-rob0t/zara#1097]
## Goal

Add an **agentic self-improvement mode** to Zara that can iteratively improve agent behavior from scored subagent trajectories while keeping Symbolic-only mode completely separate and model-free.

Research baselines:
- Microsoft Agent Lightning v1.0: harnessed agentic RL over the real deployed agent/tool/context loop.
- Microsoft SkillOpt: scored rollout -> reflect -> aggregate -> select -> bounded edit -> held-out evaluate, mutating external skill state while freezing the target model.
- Microsoft OEO: optional experimental open-ended optimizer mode under fixed objectives/budgets/evaluation.

### Symbolic only
No model inference, no RSI. Prolog + tools/search/KB only.

### Agentic
LLM/agent runtime enabled. Subagents may execute tasks, be scored, and contribute training/evolution evidence.

### Agentic + evolution
Explicit opt-in profile that allows persistent mutation of approved resources after evaluation gates.

## Core architecture

```
Task corpus / real reviewed tasks
   -> coordinator
   -> N subagent rollouts
        -> tools / search / Zara runtime / KB
        -> trajectories
   -> independent graders/verifiers
   -> score vector + evidence
   -> Prolog selection/promotion policy
   -> optimizer backend
        -> SkillOpt: mutate skill/prompt/policy artifacts
        -> Agent Lightning: train mutable model resources
        -> OEO: experimental optimizer strategy
   -> held-out validation
   -> candidate artifact/checkpoint
   -> canary
   -> promote or reject/rollback
```

## Non-negotiable invariants

- A candidate never promotes solely because it scored itself highly.
- Graders/evaluators are versioned and independently identifiable.
- Training/eval/test splits are explicit.
- Reward hacking / benchmark leakage checks are first-class.
- Every mutation has lineage: parent version, data, trajectories, scores, optimizer, resource delta, validation result.
- Mutating one resource must not silently mutate the evaluator used to approve it.
- Rollback is always possible.
- User data is excluded from training by default unless explicitly opted into an approved local/private training profile.
- No remote training upload without a capability + privacy policy permitting it.
- Agentic evolution cannot weaken Zara's Prolog permission/side-effect authority.

## Mutable resource classes

```
skill(SKILL.md / normalized Skill IR)
prompt
routing_policy
tool_description
model_weights
adapter_weights(LoRA/QLoRA)
reasoning/runtime settings
```

The selected backend advertises which resource classes are mutable.

## Local + remote

Support:
- local runner + local trainer + local model;
- local coordinator + remote inference subagents;
- local coordinator + remote training worker;
- remote/Kubernetes rollout workers;
- provider-supported remote fine-tuning/training;
- API-only frozen remote models using SkillOpt/prompt/policy evolution only.

## Completion

Close only when Zara can:
1. spawn multiple scored subagents on the same task set;
2. collect typed trajectories/rewards;
3. evolve a skill through validation-gated SkillOpt-style edits;
4. train/update at least one local open model through Agent Lightning-compatible infrastructure;
5. run a remote-worker topology;
6. reject a candidate that improves train score but loses held-out validation;
7. canary and rollback a promoted candidate;
8. expose full evolution lineage/score history in diagnostics/UI.

## Current child slices

- #1098 scored subagent actor/trajectory/reward contract
- #1099 Agent Lightning v1.x training backend
- #1100 SkillOpt SKILL.md/prompt evolution
- #1101 independent verifier + held-out anti-reward-hacking promotion gate
- #1102 local LoRA/QLoRA/RL model mutation + checkpoint rollback
- #1103 remote worker/trainer/provider-training topology
- #1104 optional OEO experimental optimizer
- #1105 evolution UI/scoreboards/lineage/canary/rollback
- #1122 automatic score-triggered multi-generation RSI loop
